book/src/getting-started/4-cmake-integration.md

```diff
@@ -49,6 +49,10 @@ We'll have to do multiple things:
 - Add `cxx`, `cxx-qt`, as well as `cxx-qt-lib` as dependencies
 - Add `cxx-qt-build` as a build dependency
 
+Note that in order for cxx-qt to work, the `qmake` executable must be located. This is because cxx-qt relies on `qmake` to locate the necessary Qt libraries and header files on your system. This can be done by either adding it to your system's `PATH`, or by setting the `QMAKE` environment variable to its path.
+
+To check which version of Qt will be used with `qmake`, you can use the `qmake -query` command. This will display information about the Qt installation, including the version number and installation path.
```

Comment:
This is helpful :+1:

Reply:
> To check which version of Qt will be used with `qmake`, you can use the `qmake -query` command. This will display information about the Qt installation, including the version number and installation path.

This isn't always correct. CMake can pick up a different version of Qt than the first `qmake` executable found in `PATH`, particularly if the user sets `CMAKE_PREFIX_PATH`.

Reply:
But we are in the cargo/build.rs section, so everything is based on the qmake exe right?

Of course from the CMake, the `find_package` as so many option and could behave in some many way. But this is not what this section is about.

Reply:
With CMake, the location that CMake chooses for `QMAKE` is passed down into Rust, so if it picked one outside of your `PATH` this should be passed into the Rust env to IIRC

I think this is done via these places
https://github.com/KDAB/cxx-qt/blob/c435d65aba8eef451596def5f29c3b52dae8ea23/examples/qml_minimal/CMakeLists.txt#L29

 https://github.com/KDAB/cxx-qt/blob/c435d65aba8eef451596def5f29c3b52dae8ea23/examples/qml_minimal/CMakeLists.txt#L51-L60

Reply:
So I think this documentation should move to the the cargo build system page.
On the cargo section in the cmake page, I shoudn't mention the PATH at all. Just QMAKE, and say that it will later be passed from whatever qt version CMake found as an env variable. Would it be ok?

Reply:
I haven't look in cargo only build yet. When you say you don't have to ahve `QMAKE` set, you still need to have it in the `PATH` right?

Reply:
I think that's how it's working there yes, but i'd have to look at the code again.

The `QMAKE` env var is for ensuring that CMake and Cargo when combined agree on the same `qmake` executable.

Reply:
Ok so on rust side it is check for `QMAKE` env variable, then fallback to `qmake` in `PATH`, then error.
In the CMake documentation, only the `QMAKE` is interesting to the user.

Reply:
Correct. The reason we must pass the QMAKE environment variable in CMake builds is to ensure that whatever Qt installation CMake picks up due to CMAKE_PREFIX_PATH or whatever other mechanism is the same Qt installation that build.rs uses.